src/main/java/com/cqse/teamscaleupload/TeamscaleUpload.java

```diff
@@ -243,17 +305,30 @@ private static void sendRequestForFormat(Input input, String format, List<String
                 .post(requestBody)
                 .build();
 
+        System.out.print("Sending upload for format " + format + "...");
+        sendRequest(input, url, request);
+    }
+
+    private static String sendRequest(Input input, HttpUrl url, Request request) throws IOException {
         OkHttpClient client = OkHttpClientUtils.createClient(input.validateSsl);
 
         try (Response response = client.newCall(request).execute()) {
             handleCommonErrors(response, input);
-            System.out.println("Upload to Teamscale successful for format " + format);
+            System.out.println("Successful");
+            return response.body().string();
+        } catch (UnknownHostException e) {
+            fail("The host " + url + " could not be resolved. Please ensure you have no typo and that" +
+                    " this host is reachable from this server. " + e.getMessage());
+        } catch (ConnectException e) {
+            fail("The URL " + url + " refused a connection. Please ensure that you have no typo and that" +
+                    " this endpoint is reachable and not blocked by firewalls. " + e.getMessage());
         } finally {
             // we must shut down OkHttp as otherwise it will leave threads running and
             // prevent JVM shutdown
             client.dispatcher().executorService().shutdownNow();
```

Comment:
this should only happen after the last request, so we can't do this in `sendRequest`, which is called multiple times. Please move this top-level into the `main` method